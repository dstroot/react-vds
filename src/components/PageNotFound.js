import React from 'react';
import PropTypes from 'prop-types';

import './PageNotFound.css';

function PageNotFound(props) {
    const {image} = props;
    return (
      <div className="FourOhFour">
        <div className="bg" style={{ backgroundImage: 'url(' + image + ')'}}></div>
        <h1 className="code">404</h1>
      </div>
    )
}

// https://codeburst.io/validating-props-easily-with-react-proptypes-96e80208207
PageNotFound.propTypes = {
  image:PropTypes.string
};

PageNotFound.defaultProps = {
  image: 'https://i.giphy.com/media/sMABNWsUWPC1y/giphy.gif'
}

export default PageNotFound;
