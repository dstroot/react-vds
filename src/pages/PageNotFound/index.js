import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

const PageNotFound = ({ image }) => {
  // Set the page title using the useEffect hook
  useEffect(() => {
    document.title = `Verite Data Science â€¢ 404`;
    requestAnimationFrame(() => {
      window.scrollTo(0, 0);
    });
  });

  return (
    <div className="hero" style={{ backgroundImage: `url(${image})` }}>
      <div className="d-flex justify-content-center color-wash">
        <h1 className="mt-5 display-3 text-white">Oops... 404</h1>
      </div>
    </div>
  );
};

// https://codeburst.io/validating-props-easily-with-react-proptypes-96e80208207
PageNotFound.propTypes = {
  image: PropTypes.string,
};

PageNotFound.defaultProps = {
  image: 'https://i.giphy.com/media/sMABNWsUWPC1y/giphy.gif',
};

export default PageNotFound;
