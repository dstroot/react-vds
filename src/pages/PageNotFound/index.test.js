import React from 'react';
import renderer from 'react-test-renderer';

import { MemoryRouter } from 'react-router-dom';
import PageNotFound from '../PageNotFound';

test('PageNotFound renders a 404 page', () => {
  const component = renderer.create(
    <MemoryRouter>
      <PageNotFound />{' '}
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // try setting the props
  const setProps = renderer.create(
    <MemoryRouter>
      <PageNotFound image="myimage.jpg" />
    </MemoryRouter>
  );
  tree = setProps.toJSON();
  expect(tree).toMatchSnapshot();
});
