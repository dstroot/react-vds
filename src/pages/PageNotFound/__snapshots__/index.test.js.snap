// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound renders a 404 page 1`] = `
Array [
  <div
    className="hero"
    style={
      Object {
        "backgroundImage": "url(https://i.giphy.com/media/sMABNWsUWPC1y/giphy.gif)",
      }
    }
  >
    <div
      className="d-flex justify-content-center color-wash"
    >
      <h1
        className="mt-5 display-3 text-white"
      >
        Oops... 404
      </h1>
    </div>
  </div>,
  " ",
]
`;

exports[`PageNotFound renders a 404 page 2`] = `
<div
  className="hero"
  style={
    Object {
      "backgroundImage": "url(myimage.jpg)",
    }
  }
>
  <div
    className="d-flex justify-content-center color-wash"
  >
    <h1
      className="mt-5 display-3 text-white"
    >
      Oops... 404
    </h1>
  </div>
</div>
`;
